Code/Tools/AssetProcessor/native/utilities/BuilderManager.cpp

```diff
@@ -433,21 +83,23 @@ namespace AssetProcessor
         {
             AZStd::lock_guard<AZStd::mutex> lock(m_buildersMutex);
 
-            AZStd::shared_ptr<Builder> builder;
-            auto itr = m_builders.find(requestPing.m_uuid);
+            auto builder = m_builderList.Find(requestPing.m_uuid);
 
-            if (itr != m_builders.end())
-            {
-                builder = itr->second;
-            }
-            else if (m_allowUnmanagedBuilderConnections)
+            if (!builder)
             {
-                AZ_TracePrintf("BuilderManager", "External builder connection accepted\n");
-                builder = AddNewBuilder();
-            }
-            else
-            {
-                AZ_Warning("BuilderManager", false, "Received request ping from builder but could not match uuid %s", requestPing.m_uuid.ToString<AZStd::string>().c_str());
+                if (m_allowUnmanagedBuilderConnections)
+                {
+                    AZ_TracePrintf("BuilderManager", "External builder connection accepted\n");
+                    builder = AddNewBuilder(BuilderPurpose::ProcessJob); // Assign it as a "process job" builder since we don't know what type it is.  This means it won't be used for create jobs
+                }
+                else
+                {
+                    AZ_Warning(
+                        "BuilderManager",
+                        false,
+                        "Received request ping from builder but could not match uuid %s",
```

Comment:
nit: "Received ping from builder, but UUID is not known so will be ignored"? Is there any action user should take to "find" or remove rogue builder?

Reply:
Expanded a bit, not sure I can recommend how to stop a "rogue" builder.